### DISCUSSION TITLE
[Idea] Improved Search

### DISCUSSION THREAD

**Discussion by sickboards** [PrestaShop/PrestaShop#37123] [Ideas]
### The problem

The default PS search seems to show the products in an order based upon position, which I think is obsolete. Also, there is little room for configuration. For me, always the least relevant products show up in the drop-down list (probably the oldest products). The search is one of the most important functions on the website, but it does not seem to meet modern day requirements.

### The solution

If the module had the possibility to configure the order (A-Z, Date Add, etc) of the drop-down list then that would already be a big win.
Ideally it would also be possible to configure whether or not to show categories, brands and/or suppliers (and how many).
 

### Alternatives

The are paid modules that do this but I believe this should be part of the default configuration.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
Prestashop search returns the products by relevance, position is only the internal name.

The relevance is configured by setting the weights for different fields in search settings.

**Comment by sickboards** [comment]
Hello Daniel
Okay but what is you have for instance 200 different vans shoes and someone
searches for vans shoes, how are the sorted as the relevance is the same
for all of them?
it would be helpful to control this behavior.
Thanks for looking into this.

**Reply by kpodemski** [reply]
and how would you control this behavior? what would you like to do to set higher priority for some of the products if the search phrase is generic like "van shoes"?

**Comment by sickboards** [comment]
Hi.
Ideally:
Weight factors (like for priority, or this can be added to the list of
existing weight factors):
Relevance, Date Add, Price, Sales, Alphabetically
Or
2 pull down menus
1) First priority (Relevance)
2) Second priority (Date add)

Thanks for getting back to me.
Martijn
